ccl/importccl: TestImportCSVStmt failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/ff04012ed2d2c0c8e30e4de106ca0a350bca8c3e

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1404933&tab=buildLog

```
W190725 05:20:00.371443 260953 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.371596 259899 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.371784 260708 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.371816 260953 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190725 05:20:00.373872 306982 storage/replica_command.go:282  [n1,s1,r388/1:/{Table/79/3/3…-Max}] initiating a split of this range at key /Table/79/3/4344/"C"/PrefixEnd [r389] (manual)
W190725 05:20:00.384070 259926 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.384244 260704 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.384267 260921 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.384648 260704 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.384648 260921 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.384658 259926 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190725 05:20:00.386965 306964 storage/replica_command.go:282  [n1,s1,r389/1:/{Table/79/3/4…-Max}] initiating a split of this range at key /Table/80 [r390] (manual)
W190725 05:20:00.398184 260703 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.398263 259904 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.398365 260930 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.398658 260703 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.398658 259904 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.398852 260930 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.420560 17 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
W190725 05:20:00.420917 17 storage/engine/rocksdb.go:116  [rocksdb] [db/version_set.cc:3086] More existing levels in DB than needed. max_bytes_for_level_multiplier may not be guaranteed.
I190725 05:20:00.440551 307713 storage/replica_command.go:282  [n2,split,s2,r342/2:/Table/7{8-9/1/741}] initiating a split of this range at key /Table/79 [r361] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-implicit-bzip (0.93s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc001f81380}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/99ba774d014955965f819d36fac7e14b2d405aa9

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1489848&tab=buildLog

```
I190916 05:16:41.335874 12996 storage/replica_command.go:410  [n1,s1,r127/1:/{Table/71/2-Max}] initiating a split of this range at key /Table/71/3 [r128] (manual)
I190916 05:16:41.351974 13123 storage/replica_command.go:410  [n1,s1,r124/1:/Table/71/{1-2}] initiating a split of this range at key /Table/71/1/2000 [r129] (manual)
I190916 05:16:41.373918 13174 storage/replica_command.go:410  [n1,s1,r127/1:/Table/71/{2-3}] initiating a split of this range at key /Table/71/2/"A"/2002 [r130] (manual)
I190916 05:16:41.386900 13243 storage/replica_command.go:410  [n1,s1,r124/1:/Table/71/1{-/2000}] initiating a split of this range at key /Table/71/1/0 [r131] (manual)
I190916 05:16:41.398352 13013 storage/replica_command.go:410  [n1,s1,r124/1:/Table/71/1{-/2000}] initiating a split of this range at key /Table/71/1/1000 [r132] (manual)
I190916 05:16:41.434233 13233 storage/replica_command.go:410  [n1,s1,r127/1:/Table/71/2{-/"A"/20…}] initiating a split of this range at key /Table/71/2/"A"/0 [r133] (manual)
I190916 05:16:41.444428 12864 storage/bulk/sst_batcher.go:383  [n3] SSTable cannot be added spanning range bounds /Table/71/2/"A"/2002, retrying...
I190916 05:16:41.463490 13311 storage/replica_command.go:410  [n3,s3,r131/3:/Table/71/1/{0-2000}] initiating a split of this range at key /Table/71/1/1000 [r70] (manual)
I190916 05:16:41.589997 13839 storage/replica_command.go:410  [n1,split,s1,r120/1:/Table/{69/3-71/1}] initiating a split of this range at key /Table/71 [r134] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-auto-gzip (0.34s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc001879000}': sql: no rows in result set
=== RUN   TestImportCSVStmt/empty-with-files
I190916 05:16:40.112354 2885 sql/event_log.go:130  [n1,client=127.0.0.1:34974,user=root] Event: "create_database", target: 62, info: {DatabaseName:csv6 Statement:CREATE DATABASE csv6 User:root}
I190916 05:16:40.136568 7322 storage/replica_command.go:410  [n1,s1,r76/1:/{Table/61/3-Max}] initiating a split of this range at key /Table/63/1 [r78] (manual)
I190916 05:16:40.136645 6659 storage/replica_command.go:410  [n1,s1,r76/1:/{Table/61/3-Max}] initiating a split of this range at key /Table/63/1 [r79] (manual)
I190916 05:16:40.138566 7321 ccl/importccl/read_import_base.go:66  [n1,import-distsql-ingest] could not fetch file size; falling back to per-file progress: <nil>
I190916 05:16:40.144712 7370 storage/replica_command.go:410  [n1,s1,r79/1:/{Table/63/1-Max}] initiating a split of this range at key /Table/63/2 [r81] (manual)
I190916 05:16:40.164676 7441 storage/replica_command.go:410  [n1,s1,r81/1:/{Table/63/2-Max}] initiating a split of this range at key /Table/63/3 [r82] (manual)
I190916 05:16:40.179305 7500 storage/replica_command.go:410  [n1,s1,r79/1:/Table/63/{1-2}] initiating a split of this range at key /Table/63/1/1000 [r83] (manual)
I190916 05:16:40.212228 7491 storage/replica_command.go:410  [n1,s1,r81/1:/Table/63/{2-3}] initiating a split of this range at key /Table/63/2/"A"/1014 [r84] (manual)
I190916 05:16:40.286320 7322 storage/replica_command.go:410  [n1,s1,r83/1:/Table/63/{1/1000-2}] initiating a split of this range at key /Table/63/1/2000 [r85] (manual)
I190916 05:16:40.310834 7639 storage/replica_command.go:410  [n2,s2,r84/2:/Table/63/{2/"A"/1…-3}] initiating a split of this range at key /Table/63/2/"A"/2002 [r46] (manual)
I190916 05:16:40.338928 7297 storage/replica_command.go:410  [n1,s1,r76/1:/Table/6{1/3-3/1}] initiating a split of this range at key /Table/63/1 [r80] (manual); delayed split for 0.2s to avoid Raft snapshot
I190916 05:16:40.390891 7736 storage/replica_command.go:410  [n1,s1,r79/1:/Table/63/1{-/1000}] initiating a split of this range at key /Table/63/1/0 [r86] (manual)
I190916 05:16:40.485243 7995 storage/replica_command.go:410  [n1,split,s1,r76/1:/Table/6{1/3-3/1}] initiating a split of this range at key /Table/63 [r87] (zone config)
    --- FAIL: TestImportCSVStmt/empty-with-files (0.39s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc002b36d80}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/250f4c36de2b88eff443cf9be9cd5d2759312c88

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1512323&tab=buildLog

```
I190928 05:16:40.700387 2973 storage/replica_command.go:411  [n1,s1,r26/1:/{Table/54/2-Max}] initiating a split of this range at key /Table/54/3 [r27] (manual)
I190928 05:16:40.743975 3605 storage/replica_command.go:411  [n1,s1,r25/1:/Table/54/{1-2}] initiating a split of this range at key /Table/54/1/2000 [r28] (manual)
I190928 05:16:40.744891 3560 storage/replica_command.go:411  [n1,s1,r25/1:/Table/54/{1-2}] initiating a split of this range at key /Table/54/1/0 [r29] (manual)
I190928 05:16:40.761235 3587 storage/replica_command.go:411  [n1,s1,r25/1:/Table/54/1{-/2000}] initiating a split of this range at key /Table/54/1/1000 [r30] (manual)
I190928 05:16:40.856172 3560 storage/replica_command.go:411  [n1,s1,r25/1:/Table/54/1{-/1000}] initiating a split of this range at key /Table/54/1/0 [r31] (manual)
I190928 05:16:40.909172 3681 storage/replica_command.go:411  [n1,s1,r26/1:/Table/54/{2-3}] initiating a split of this range at key /Table/54/2/"A"/1014 [r32] (manual)
I190928 05:16:40.938615 3560 storage/replica_command.go:411  [n1,s1,r26/1:/Table/54/2{-/"A"/10…}] initiating a split of this range at key /Table/54/2/"A"/0 [r33] (manual)
I190928 05:16:40.948172 3560 storage/bulk/sst_batcher.go:383  [n1,import-distsql-ingest] SSTable cannot be added spanning range bounds /Table/54/2/"A"/1014, retrying...
I190928 05:16:40.976624 3772 storage/replica_command.go:411  [n1,s1,r32/1:/Table/54/{2/"A"/1…-3}] initiating a split of this range at key /Table/54/2/"A"/2002 [r34] (manual)
I190928 05:16:41.082930 3881 storage/replica_command.go:411  [n1,split,s1,r24/1:/Table/{28-54/1}] initiating a split of this range at key /Table/54 [r35] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-intodb (0.44s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc001852280}': sql: no rows in result set
=== RUN   TestImportCSVStmt/schema-in-query-opts
I190928 05:16:43.402706 2900 sql/event_log.go:130  [n1,client=127.0.0.1:52126,user=root] Event: "create_database", target: 57, info: {DatabaseName:csv3 Statement:CREATE DATABASE csv3 User:root}
I190928 05:16:43.632339 5463 storage/replica_command.go:411  [n1,s1,r38/1:/{Table/56/3-Max}] initiating a split of this range at key /Table/58/1 [r49] (manual); delayed split for 0.2s to avoid Raft snapshot
I190928 05:16:43.640212 5463 storage/replica_command.go:411  [n1,s1,r49/1:/{Table/58/1-Max}] initiating a split of this range at key /Table/58/2 [r50] (manual)
I190928 05:16:43.654569 5463 storage/replica_command.go:411  [n1,s1,r50/1:/{Table/58/2-Max}] initiating a split of this range at key /Table/58/3 [r51] (manual)
I190928 05:16:43.682139 5680 storage/replica_command.go:411  [n1,s1,r49/1:/Table/58/{1-2}] initiating a split of this range at key /Table/58/1/2000 [r52] (manual)
I190928 05:16:43.694253 5538 storage/replica_command.go:411  [n1,s1,r49/1:/Table/58/{1-2}] initiating a split of this range at key /Table/58/1/1000 [r53] (manual)
I190928 05:16:43.696519 5753 storage/replica_command.go:411  [n1,s1,r49/1:/Table/58/1{-/2000}] initiating a split of this range at key /Table/58/1/0 [r54] (manual)
I190928 05:16:43.716097 5680 storage/replica_command.go:411  [n1,s1,r50/1:/Table/58/{2-3}] initiating a split of this range at key /Table/58/2/NULL/2000 [r65] (manual)
I190928 05:16:43.750277 5667 storage/replica_command.go:411  [n1,s1,r50/1:/Table/58/2{-/NULL/2…}] initiating a split of this range at key /Table/58/2/NULL/0 [r66] (manual)
I190928 05:16:43.750582 5790 storage/replica_command.go:411  [n2,s2,r54/3:/Table/58/1/{0-2000}] initiating a split of this range at key /Table/58/1/1000 [r75] (manual)
I190928 05:16:43.760039 5630 storage/bulk/sst_batcher.go:383  [n2] SSTable cannot be added spanning range bounds /Table/58/2/NULL/2000, retrying...
I190928 05:16:43.895007 6233 storage/replica_command.go:411  [n1,split,s1,r38/1:/Table/5{6/3-8/1}] initiating a split of this range at key /Table/58 [r67] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-query-opts (0.50s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc000445c00}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/1ed03c9811f01fef31950a5cb75a7b591af6fc26

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1518477&tab=buildLog

```
I191003 05:22:46.142745 14400 storage/replica_command.go:411  [n1,s1,r132/1:/Table/72/{1-2}] initiating a split of this range at key /Table/72/1/2000 [r135] (manual)
I191003 05:22:46.148388 14404 storage/replica_command.go:411  [n1,s1,r132/1:/Table/72/1{-/2000}] initiating a split of this range at key /Table/72/1/1000 [r136] (manual)
I191003 05:22:46.150826 14534 storage/replica_command.go:411  [n1,s1,r132/1:/Table/72/1{-/2000}] initiating a split of this range at key /Table/72/1/0 [r137] (manual)
I191003 05:22:46.156028 14571 storage/replica_command.go:411  [n1,s1,r133/1:/Table/72/{2-3}] initiating a split of this range at key /Table/72/2/"A"/2002 [r138] (manual)
I191003 05:22:46.170444 14534 storage/replica_command.go:411  [n1,s1,r133/1:/Table/72/2{-/"A"/20…}] initiating a split of this range at key /Table/72/2/"A"/0 [r139] (manual)
I191003 05:22:46.174885 14534 storage/bulk/sst_batcher.go:394  [n1,import-distsql-ingest] SSTable cannot be added spanning range bounds /Table/72/2/"A"/2002, retrying...
I191003 05:22:46.198786 14669 storage/replica_command.go:411  [n1,s1,r137/1:/Table/72/1/{0-2000}] initiating a split of this range at key /Table/72/1/1000 [r140] (manual)
I191003 05:22:46.254731 15397 storage/replica_command.go:411  [n1,split,s1,r123/1:/Table/7{0/3-2/1}] initiating a split of this range at key /Table/72 [r141] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-auto-gzip (0.19s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc0037d9e00}': sql: no rows in result set
=== RUN   TestImportCSVStmt/schema-in-file-auto-bzip
I191003 05:22:46.697572 2880 sql/event_log.go:130  [n1,client=127.0.0.1:55276,user=root] Event: "create_database", target: 77, info: {DatabaseName:csv13 Statement:CREATE DATABASE csv13 User:root}
I191003 05:22:46.717943 19765 storage/replica_command.go:411  [n1,s1,r152/1:/{Table/76/3-Max}] initiating a split of this range at key /Table/78/1 [r169] (manual)
I191003 05:22:46.723772 19765 storage/replica_command.go:411  [n1,s1,r169/1:/{Table/78/1-Max}] initiating a split of this range at key /Table/78/2 [r170] (manual)
I191003 05:22:46.734078 19765 storage/replica_command.go:411  [n1,s1,r170/1:/{Table/78/2-Max}] initiating a split of this range at key /Table/78/3 [r171] (manual)
I191003 05:22:46.762341 19941 storage/replica_command.go:411  [n1,s1,r169/1:/Table/78/{1-2}] initiating a split of this range at key /Table/78/1/2000 [r172] (manual)
I191003 05:22:46.763394 19908 storage/replica_command.go:411  [n1,s1,r169/1:/Table/78/{1-2}] initiating a split of this range at key /Table/78/1/0 [r173] (manual)
I191003 05:22:46.764134 19796 storage/replica_command.go:411  [n1,s1,r169/1:/Table/78/{1-2}] initiating a split of this range at key /Table/78/1/1000 [r174] (manual)
I191003 05:22:46.781499 19949 storage/replica_command.go:411  [n1,s1,r170/1:/Table/78/{2-3}] initiating a split of this range at key /Table/78/2/"A"/0 [r175] (manual)
I191003 05:22:46.827046 20009 storage/replica_command.go:411  [n3,s3,r173/2:/Table/78/{1/0-2}] initiating a split of this range at key /Table/78/1/1000 [r100] (manual)
I191003 05:22:46.828076 20012 storage/replica_command.go:411  [n3,s3,r173/2:/Table/78/{1/0-2}] initiating a split of this range at key /Table/78/1/2000 [r101] (manual)
I191003 05:22:46.851381 20102 storage/replica_command.go:411  [n3,s3,r175/2:/Table/78/{2/"A"/0-3}] initiating a split of this range at key /Table/78/2/"A"/1014 [r102] (manual)
I191003 05:22:46.893194 19885 storage/replica_command.go:411  [n2,s2,r100/3:/Table/78/{1/1000-2}] initiating a split of this range at key /Table/78/1/2000 [r47] (manual)
I191003 05:22:46.920244 20132 storage/replica_command.go:411  [n1,s1,r102/1:/Table/78/{2/"A"/1…-3}] initiating a split of this range at key /Table/78/2/"A"/2002 [r176] (manual)
I191003 05:22:46.967028 20307 storage/replica_command.go:411  [n1,split,s1,r152/1:/Table/7{6/3-8/1}] initiating a split of this range at key /Table/78 [r177] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-auto-bzip (0.29s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc003800380}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/236862d9fc5b23f14aab920fa7b8334649ba8be3

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1524173&tab=buildLog

```
=== RUN   TestImportCSVStmt/schema-in-file-auto-bzip
I191007 05:14:37.111975 2865 sql/event_log.go:130  [n1,client=127.0.0.1:52148,user=root] Event: "create_database", target: 76, info: {DatabaseName:csv13 Statement:CREATE DATABASE csv13 User:root}
I191007 05:14:37.135897 18111 storage/replica_command.go:411  [n1,s1,r156/1:/{Table/75/3-Max}] initiating a split of this range at key /Table/77/1 [r162] (manual)
I191007 05:14:37.151090 18111 storage/replica_command.go:411  [n1,s1,r162/1:/{Table/77/1-Max}] initiating a split of this range at key /Table/77/2 [r163] (manual)
I191007 05:14:37.163923 18111 storage/replica_command.go:411  [n1,s1,r163/1:/{Table/77/2-Max}] initiating a split of this range at key /Table/77/3 [r164] (manual)
I191007 05:14:37.191560 18422 storage/replica_command.go:411  [n1,s1,r162/1:/Table/77/{1-2}] initiating a split of this range at key /Table/77/1/2000 [r165] (manual)
I191007 05:14:37.192427 18447 storage/replica_command.go:411  [n1,s1,r162/1:/Table/77/{1-2}] initiating a split of this range at key /Table/77/1/0 [r166] (manual)
I191007 05:14:37.195800 18051 storage/replica_command.go:411  [n1,s1,r162/1:/Table/77/{1-2}] initiating a split of this range at key /Table/77/1/1000 [r167] (manual)
I191007 05:14:37.216329 18428 storage/replica_command.go:411  [n1,s1,r163/1:/Table/77/{2-3}] initiating a split of this range at key /Table/77/2/"A"/2002 [r168] (manual)
I191007 05:14:37.261801 18051 storage/replica_command.go:411  [n1,s1,r162/1:/Table/77/1{-/2000}] initiating a split of this range at key /Table/77/1/1000 [r169] (manual)
I191007 05:14:37.265873 18447 storage/replica_command.go:411  [n1,s1,r162/1:/Table/77/1{-/2000}] initiating a split of this range at key /Table/77/1/0 [r170] (manual)
I191007 05:14:37.386505 18447 storage/replica_command.go:411  [n1,s1,r162/1:/Table/77/1{-/1000}] initiating a split of this range at key /Table/77/1/0 [r171] (manual)
I191007 05:14:37.461859 19281 storage/replica_command.go:411  [n1,split,s1,r156/1:/Table/7{5/3-7/1}] initiating a split of this range at key /Table/77 [r172] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-auto-bzip (0.36s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc002e9d900}': sql: no rows in result set
=== RUN   TestImportCSVStmt/schema-in-file-implicit-bzip
I191007 05:14:37.474887 2865 sql/event_log.go:130  [n1,client=127.0.0.1:52148,user=root] Event: "create_database", target: 78, info: {DatabaseName:csv14 Statement:CREATE DATABASE csv14 User:root}
I191007 05:14:37.500756 20248 storage/replica_command.go:411  [n1,s1,r164/1:/{Table/77/3-Max}] initiating a split of this range at key /Table/79/1 [r173] (manual)
I191007 05:14:37.509200 20248 storage/replica_command.go:411  [n1,s1,r173/1:/{Table/79/1-Max}] initiating a split of this range at key /Table/79/2 [r174] (manual)
I191007 05:14:37.522091 20248 storage/replica_command.go:411  [n1,s1,r174/1:/{Table/79/2-Max}] initiating a split of this range at key /Table/79/3 [r175] (manual)
I191007 05:14:37.558200 20269 storage/replica_command.go:411  [n1,s1,r173/1:/Table/79/{1-2}] initiating a split of this range at key /Table/79/1/2000 [r176] (manual)
I191007 05:14:37.571025 20333 storage/replica_command.go:411  [n1,s1,r173/1:/Table/79/{1-2}] initiating a split of this range at key /Table/79/1/1000 [r177] (manual)
I191007 05:14:37.575413 20334 storage/replica_command.go:411  [n1,s1,r173/1:/Table/79/1{-/2000}] initiating a split of this range at key /Table/79/1/0 [r178] (manual)
I191007 05:14:37.583066 20269 storage/replica_command.go:411  [n1,s1,r174/1:/Table/79/{2-3}] initiating a split of this range at key /Table/79/2/"A"/2002 [r179] (manual)
I191007 05:14:37.634123 20349 storage/replica_command.go:411  [n2,s2,r178/2:/Table/79/1/{0-2000}] initiating a split of this range at key /Table/79/1/1000 [r61] (manual)
I191007 05:14:37.713633 21176 storage/replica_command.go:411  [n1,split,s1,r164/1:/Table/7{7/3-9/1}] initiating a split of this range at key /Table/79 [r180] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-implicit-bzip (0.25s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc001f85600}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/c0cab6082c2383e77f0a806c10a1506b6aa7b6bf

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1526024&tab=buildLog

```
I191008 05:16:21.353216 20314 storage/replica_command.go:411  [n1,s1,r170/1:/Table/79/{1-2}] initiating a split of this range at key /Table/79/1/2000 [r173] (manual)
I191008 05:16:21.357103 20361 storage/replica_command.go:411  [n1,s1,r170/1:/Table/79/{1-2}] initiating a split of this range at key /Table/79/1/1000 [r174] (manual)
I191008 05:16:21.359031 20224 storage/replica_command.go:411  [n1,s1,r170/1:/Table/79/{1-2}] initiating a split of this range at key /Table/79/1/0 [r175] (manual)
I191008 05:16:21.368243 20353 storage/replica_command.go:411  [n1,s1,r171/1:/Table/79/{2-3}] initiating a split of this range at key /Table/79/2/"A"/2002 [r176] (manual)
I191008 05:16:21.415346 20224 storage/replica_command.go:411  [n1,s1,r170/1:/Table/79/1{-/2000}] initiating a split of this range at key /Table/79/1/0 [r177] (manual)
I191008 05:16:21.416274 20361 storage/replica_command.go:411  [n1,s1,r170/1:/Table/79/1{-/2000}] initiating a split of this range at key /Table/79/1/1000 [r178] (manual)
I191008 05:16:21.541049 20459 storage/replica_command.go:411  [n1,s1,r177/1:/Table/79/1/{0-2000}] initiating a split of this range at key /Table/79/1/1000 [r179] (manual)
I191008 05:16:21.592019 20581 storage/replica_command.go:411  [n1,split,s1,r152/1:/Table/7{7/3-9/1}] initiating a split of this range at key /Table/79 [r180] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-implicit-bzip (0.32s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc0017f2f00}': sql: no rows in result set
=== RUN   TestImportCSVStmt/schema-in-file-no-decompress
I191008 05:16:19.617898 2918 sql/event_log.go:130  [n1,client=127.0.0.1:40154,user=root] Event: "create_database", target: 66, info: {DatabaseName:csv8 Statement:CREATE DATABASE csv8 User:root}
I191008 05:16:19.645748 9007 storage/replica_command.go:411  [n1,s1,r91/1:/{Table/65/3-Max}] initiating a split of this range at key /Table/67/1 [r99] (manual)
I191008 05:16:19.653816 9007 storage/replica_command.go:411  [n1,s1,r99/1:/{Table/67/1-Max}] initiating a split of this range at key /Table/67/2 [r100] (manual)
I191008 05:16:19.665479 9007 storage/replica_command.go:411  [n1,s1,r100/1:/{Table/67/2-Max}] initiating a split of this range at key /Table/67/3 [r101] (manual)
I191008 05:16:19.719940 10043 storage/replica_command.go:411  [n1,s1,r99/1:/Table/67/{1-2}] initiating a split of this range at key /Table/67/1/0 [r102] (manual)
I191008 05:16:19.720569 10137 storage/replica_command.go:411  [n1,s1,r99/1:/Table/67/{1-2}] initiating a split of this range at key /Table/67/1/1000 [r103] (manual)
I191008 05:16:19.720891 10138 storage/replica_command.go:411  [n1,s1,r99/1:/Table/67/{1-2}] initiating a split of this range at key /Table/67/1/2000 [r104] (manual)
I191008 05:16:19.739567 10043 storage/replica_command.go:411  [n1,s1,r100/1:/Table/67/{2-3}] initiating a split of this range at key /Table/67/2/"A"/0 [r105] (manual)
I191008 05:16:19.779534 10261 storage/replica_command.go:411  [n1,s1,r102/1:/Table/67/{1/0-2}] initiating a split of this range at key /Table/67/1/2000 [r106] (manual)
I191008 05:16:19.786013 10232 storage/replica_command.go:411  [n1,s1,r102/1:/Table/67/1/{0-2000}] initiating a split of this range at key /Table/67/1/1000 [r107] (manual)
I191008 05:16:19.799902 10280 storage/replica_command.go:411  [n1,s1,r105/1:/Table/67/{2/"A"/0-3}] initiating a split of this range at key /Table/67/2/"A"/2002 [r108] (manual)
I191008 05:16:19.851833 10307 storage/replica_command.go:411  [n1,s1,r105/1:/Table/67/2/"A"/{0-2002}] initiating a split of this range at key /Table/67/2/"A"/1014 [r109] (manual)
I191008 05:16:19.863984 10107 storage/bulk/sst_batcher.go:394  [n3] SSTable cannot be added spanning range bounds /Table/67/2/"A"/2002, retrying...
I191008 05:16:19.934398 10721 storage/replica_command.go:411  [n1,split,s1,r91/1:/Table/6{5/3-7/1}] initiating a split of this range at key /Table/67 [r110] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-no-decompress (0.33s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc002d48a00}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/b9dc16b9d6cbd4627bac4b6329e87e29e52401b9

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1571342&tab=buildLog

```
I191103 06:16:29.657774 5023 storage/replica_command.go:411  [n1,s1,r55/1:/Table/58/{1-2}] initiating a split of this range at key /Table/58/1/2000 [r58] (manual)
I191103 06:16:29.668711 5123 storage/replica_command.go:411  [n1,s1,r55/1:/Table/58/1{-/2000}] initiating a split of this range at key /Table/58/1/1000 [r59] (manual)
I191103 06:16:29.676299 5184 storage/replica_command.go:411  [n1,s1,r55/1:/Table/58/1{-/2000}] initiating a split of this range at key /Table/58/1/0 [r60] (manual)
I191103 06:16:29.689232 5231 storage/replica_command.go:411  [n1,s1,r56/1:/Table/58/{2-3}] initiating a split of this range at key /Table/58/2/NULL/2000 [r61] (manual)
I191103 06:16:29.695484 5123 storage/replica_command.go:411  [n1,s1,r56/1:/Table/58/2{-/NULL/2…}] initiating a split of this range at key /Table/58/2/NULL/1000 [r62] (manual)
I191103 06:16:29.706899 5123 storage/bulk/sst_batcher.go:419  [n1,import-distsql-ingest] SSTable cannot be added spanning range bounds /Table/58/2/NULL/2000, retrying...
I191103 06:16:29.723163 958 gossip/gossip.go:1531  [n2] node has connected to cluster via gossip
I191103 06:16:29.723305 958 storage/stores.go:259  [n2] wrote 2 node addresses to persistent storage
I191103 06:16:29.754135 5184 storage/replica_command.go:411  [n1,s1,r55/1:/Table/58/1{-/1000}] initiating a split of this range at key /Table/58/1/0 [r63] (manual)
I191103 06:16:29.804609 5514 storage/replica_command.go:411  [n1,split,s1,r36/1:/Table/5{6/3-8/1}] initiating a split of this range at key /Table/58 [r64] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-query-opts (0.22s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc002afd700}': sql: no rows in result set
=== RUN   TestImportCSVStmt/schema-in-file-auto-bzip
I191103 06:16:31.641044 2655 sql/event_log.go:132  [n1,client=127.0.0.1:39172,user=root] Event: "create_database", target: 77, info: {DatabaseName:csv13 Statement:CREATE DATABASE csv13 User:root}
I191103 06:16:31.662014 19015 storage/replica_command.go:411  [n1,s1,r151/1:/{Table/76/3-Max}] initiating a split of this range at key /Table/78/1 [r158] (manual)
I191103 06:16:31.667772 19015 storage/replica_command.go:411  [n1,s1,r158/1:/{Table/78/1-Max}] initiating a split of this range at key /Table/78/2 [r159] (manual)
I191103 06:16:31.678196 19015 storage/replica_command.go:411  [n1,s1,r159/1:/{Table/78/2-Max}] initiating a split of this range at key /Table/78/3 [r160] (manual)
I191103 06:16:31.707126 19157 storage/replica_command.go:411  [n1,s1,r158/1:/Table/78/{1-2}] initiating a split of this range at key /Table/78/1/1000 [r161] (manual)
I191103 06:16:31.707794 19117 storage/replica_command.go:411  [n1,s1,r158/1:/Table/78/{1-2}] initiating a split of this range at key /Table/78/1/2000 [r162] (manual)
I191103 06:16:31.711801 19162 storage/replica_command.go:411  [n1,s1,r158/1:/Table/78/1{-/1000}] initiating a split of this range at key /Table/78/1/0 [r163] (manual)
I191103 06:16:31.724100 19206 storage/replica_command.go:411  [n1,s1,r159/1:/Table/78/{2-3}] initiating a split of this range at key /Table/78/2/"A"/1014 [r164] (manual)
I191103 06:16:31.735979 19222 storage/replica_command.go:411  [n1,s1,r159/1:/Table/78/2{-/"A"/10…}] initiating a split of this range at key /Table/78/2/"A"/0 [r165] (manual)
I191103 06:16:31.745357 19100 storage/bulk/sst_batcher.go:419  [n2] SSTable cannot be added spanning range bounds /Table/78/2/"A"/1014, retrying...
I191103 06:16:31.779328 19117 storage/replica_command.go:411  [n1,s1,r161/1:/Table/78/{1/1000-2}] initiating a split of this range at key /Table/78/1/2000 [r166] (manual)
I191103 06:16:31.792646 19117 storage/replica_command.go:411  [n1,s1,r164/1:/Table/78/{2/"A"/1…-3}] initiating a split of this range at key /Table/78/2/"A"/2002 [r167] (manual)
I191103 06:16:31.826677 20076 storage/replica_command.go:411  [n1,split,s1,r151/1:/Table/7{6/3-8/1}] initiating a split of this range at key /Table/78 [r168] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-auto-bzip (0.20s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc003baf400}': sql: no rows in result set

```

SHA: https://github.com/cockroachdb/cockroach/commits/1c3fce73cd25fa69e5a8c05ce8e215d66b4f49e4

Parameters:
```
TAGS=
GOFLAGS=-parallel=4
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestImportCSVStmt PKG=github.com/cockroachdb/cockroach/pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1599898&tab=buildLog

```
I191120 06:21:12.703961 7481 storage/replica_command.go:413  [n1,s1,r54/1:/Table/60/1{-/2000}] initiating a split of this range at key /Table/60/1/0 [r69] (manual)
I191120 06:21:12.715090 7543 storage/replica_command.go:413  [n1,s1,r65/1:/Table/60/{2-3}] initiating a split of this range at key /Table/60/2/"A"/2002 [r70] (manual)
I191120 06:21:12.731129 7481 storage/replica_command.go:413  [n1,s1,r54/1:/Table/60/1{-/2000}] initiating a split of this range at key /Table/60/1/0 [r71] (manual)
I191120 06:21:12.753611 7640 storage/replica_command.go:413  [n1,s1,r71/1:/Table/60/1/{0-2000}] initiating a split of this range at key /Table/60/1/1000 [r72] (manual)
I191120 06:21:12.797563 7773 storage/replica_command.go:413  [n1,split,s1,r48/1:/Table/{58/3-60/1}] initiating a split of this range at key /Table/60 [r73] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-sstsize (0.17s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc000eada00}': sql: no rows in result set
=== RUN   TestImportCSVStmt/empty-file
I191120 06:21:12.807110 3261 sql/event_log.go:132  [n1,client=127.0.0.1:59170,user=root] Event: "create_database", target: 61, info: {DatabaseName:csv5 Statement:CREATE DATABASE csv5 User:root}
I191120 06:21:12.828008 8348 storage/replica_command.go:413  [n1,s1,r66/1:/{Table/60/3-Max}] initiating a split of this range at key /Table/62/1 [r74] (manual)
I191120 06:21:12.832946 8348 storage/replica_command.go:413  [n1,s1,r74/1:/{Table/62/1-Max}] initiating a split of this range at key /Table/62/2 [r75] (manual)
I191120 06:21:12.847744 8348 storage/replica_command.go:413  [n1,s1,r75/1:/{Table/62/2-Max}] initiating a split of this range at key /Table/62/3 [r76] (manual)
I191120 06:21:12.857748 8478 ccl/importccl/read_import_base.go:144  [n1,import-distsql-ingest] could not fetch file size; falling back to per-file progress: <nil>
I191120 06:21:12.862868 8755 storage/replica_command.go:413  [n1,split,s1,r66/1:/Table/6{0/3-2/1}] initiating a split of this range at key /Table/62 [r77] (zone config)
    --- FAIL: TestImportCSVStmt/empty-file (0.07s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc001040600}': sql: no rows in result set
=== RUN   TestImportCSVStmt/schema-in-file-explicit-gzip
I191120 06:21:13.563646 3261 sql/event_log.go:132  [n1,client=127.0.0.1:59170,user=root] Event: "create_database", target: 69, info: {DatabaseName:csv9 Statement:CREATE DATABASE csv9 User:root}
I191120 06:21:13.580279 13612 storage/replica_command.go:413  [n1,s1,r101/1:/{Table/68/3-Max}] initiating a split of this range at key /Table/70/1 [r109] (manual)
I191120 06:21:13.585138 13612 storage/replica_command.go:413  [n1,s1,r109/1:/{Table/70/1-Max}] initiating a split of this range at key /Table/70/2 [r110] (manual)
I191120 06:21:13.600096 13612 storage/replica_command.go:413  [n1,s1,r110/1:/{Table/70/2-Max}] initiating a split of this range at key /Table/70/3 [r111] (manual)
I191120 06:21:13.614034 13801 storage/replica_command.go:413  [n1,s1,r109/1:/Table/70/{1-2}] initiating a split of this range at key /Table/70/1/2000 [r112] (manual)
I191120 06:21:13.617698 13817 storage/replica_command.go:413  [n1,s1,r109/1:/Table/70/1{-/2000}] initiating a split of this range at key /Table/70/1/1000 [r113] (manual)
I191120 06:21:13.618042 13837 storage/replica_command.go:413  [n1,s1,r109/1:/Table/70/1{-/2000}] initiating a split of this range at key /Table/70/1/0 [r114] (manual)
I191120 06:21:13.624781 13801 storage/replica_command.go:413  [n1,s1,r110/1:/Table/70/{2-3}] initiating a split of this range at key /Table/70/2/"A"/2002 [r125] (manual)
I191120 06:21:13.665155 13837 storage/replica_command.go:413  [n1,s1,r109/1:/Table/70/1{-/1000}] initiating a split of this range at key /Table/70/1/0 [r126] (manual)
I191120 06:21:13.716051 14463 storage/replica_command.go:413  [n1,split,s1,r101/1:/Table/{68/3-70/1}] initiating a split of this range at key /Table/70 [r127] (zone config)
    --- FAIL: TestImportCSVStmt/schema-in-file-explicit-gzip (0.16s)
        jobs_verification.go:139: error scanning '&{<nil> 0xc0023dd580}': sql: no rows in result set

```

[(ccl/importccl).TestImportCSVStmt failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1713136&tab=buildLog) on [master@7a38d441b96cba33895259d22a5a2df22ca71de8](https://github.com/cockroachdb/cockroach/commits/7a38d441b96cba33895259d22a5a2df22ca71de8):

Fatal error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x26ee981]

```

Stack:
```
goroutine 72679 [running]:
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).CreateAndStartJob(0xc00207e790, 0x69d3160, 0xc0031deac0, 0xc003661e00, 0xc000ee9ad0, 0x2b, 0xc001323f80, 0x59, 0x5cfcec2, 0x4, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:218 +0x4d1
github.com/cockroachdb/cockroach/pkg/sql.(*createStatsNode).startJob(0xc006d81810, 0x69d3160, 0xc0031deac0, 0xc003661e00, 0xc0024bdd60, 0xc0024bd390)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/create_stats.go:122 +0x282
github.com/cockroachdb/cockroach/pkg/sql.(*createStatsNode).startExec.func1(0xc006d81810, 0x69d3160, 0xc0031deac0, 0xc001d9b900, 0xc0024bd398)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/create_stats.go:78 +0x90
created by github.com/cockroachdb/cockroach/pkg/sql.(*createStatsNode).startExec
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/create_stats.go:77 +0x122

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestImportCSVStmt/allow-default/2,5,e,-1,,
I200127 19:26:25.896010 4683 sql/schema_changer.go:967  [n2,scExec] schema change on t (104 v2) mutation 0 starting execution...
I200127 19:26:25.951686 3713 sql/schema_changer.go:967  [n1,scExec] schema change on t (104 v2) mutation 0 starting execution...
I200127 19:26:26.270439 3713 sql/schema_changer.go:967  [n1,scExec] schema change on t (104 v2) mutation 0 starting execution...
I200127 19:26:26.314973 4683 sql/schema_changer.go:967  [n2,scExec] schema change on t (104 v2) mutation 0 starting execution...
I200127 19:26:26.561156 69910 storage/replica_command.go:406  [n1,split,s1,r117/1:/Table/10{3/1-4/1}] initiating a split of this range at key /Table/104 [r120] (zone config)
I200127 19:26:26.873492 4830 sql/schema_changer.go:1260  [n3,scExec] waiting for a single version of table 104...
I200127 19:26:26.893808 4830 sql/schema_changer.go:1262  [n3,scExec] waiting for a single version of table 104... done (at v 0)
W200127 19:26:26.894033 4830 sql/schema_changer.go:2075  [n3] Error executing schema change: descriptor not found
I200127 19:26:27.245862 70164 storage/replica_command.go:672  [n1,merge,s1,r111/1:/Table/10{0/1-1}] initiating a merge of r114:/Table/101{-/1} [(n1,s1):1, (n3,s3):2, (n2,s2):3, next=4, gen=67] into this range (lhs+rhs has (size=0 B+0 B qps=0.00+0.00 --> 0.00qps) below threshold (size=0 B, qps=0.00))
I200127 19:26:27.287726 69928 jobs/registry.go:641  [n1] job 524559270286884865: stepping through state running
I200127 19:26:28.508610 3650 storage/store_remove_replica.go:129  [n1,s1,r111/1:/Table/10{0/1-1}] removing replica r114/1
I200127 19:26:28.519466 4550 storage/store_remove_replica.go:129  [n2,s2,r111/3:/Table/10{0/1-1}] removing replica r114/3
I200127 19:26:28.558225 4950 storage/store_remove_replica.go:129  [n3,s3,r111/2:/Table/10{0/1-1}] removing replica r114/2
I200127 19:26:28.982682 70879 jobs/registry.go:641  [n1] job 524559273107423233: stepping through state running
I200127 19:26:29.370964 3711 server/status/runtime.go:498  [n1] runtime stats: 1.3 GiB RSS, 796 goroutines, 78 MiB/95 MiB/134 MiB GO alloc/idle/total, 176 MiB/233 MiB CGO alloc/total, 1005.3 CGO/sec, 157.6/16.2 %(u/s)time, 0.9 %gc (7x), 3.6 MiB/3.6 MiB (r/w)net
I200127 19:26:30.557811 4681 server/status/runtime.go:498  [n2] runtime stats: 1.3 GiB RSS, 802 goroutines, 72 MiB/99 MiB/135 MiB GO alloc/idle/total, 176 MiB/233 MiB CGO alloc/total, 1016.4 CGO/sec, 153.9/16.1 %(u/s)time, 0.8 %gc (7x), 3.6 MiB/3.6 MiB (r/w)net
I200127 19:26:31.449137 70879 storage/replica_command.go:406  [n1,s1,r119/1:/{Table/104/1-Max}] initiating a split of this range at key /Table/105/1 [r121] (manual)
I200127 19:26:31.508458 4828 server/status/runtime.go:498  [n3] runtime stats: 1.3 GiB RSS, 803 goroutines, 64 MiB/107 MiB/135 MiB GO alloc/idle/total, 176 MiB/233 MiB CGO alloc/total, 1031.7 CGO/sec, 157.7/15.2 %(u/s)time, 0.9 %gc (8x), 3.6 MiB/3.6 MiB (r/w)net
I200127 19:26:32.836905 71926 ccl/importccl/read_import_base.go:144  [n2] could not fetch file size; falling back to per-file progress: bad ContentLength: -1
I200127 19:26:33.786347 69928 jobs/registry.go:641  [n1] job 524559270286884865: stepping through state succeeded
I200127 19:26:34.359674 70879 jobs/registry.go:641  [n1] job 524559273107423233: stepping through state succeeded
I200127 19:26:34.474408 5015 storage/queue.go:524  [n3,s3] rate limited in MaybeAdd (merge): throttled on async limiting semaphore
I200127 19:26:35.220094 73043 storage/replica_command.go:406  [n1,split,s1,r119/1:/Table/10{4/1-5/1}] initiating a split of this range at key /Table/105 [r122] (zone config)
I200127 19:26:35.874143 3404 util/stop/stopper.go:539  quiescing
I200127 19:26:35.878596 73205 util/stop/stopper.go:539  quiescing
I200127 19:26:35.886593 73204 util/stop/stopper.go:539  quiescing
E200127 19:26:35.894565 4913 jobs/registry.go:376  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
W200127 19:26:35.900854 5134 storage/raft_transport.go:637  [n2] while processing outgoing Raft queue to node 1: EOF:
W200127 19:26:35.921786 4667 storage/store.go:1526  [n2,s2,r1/2:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I200127 19:26:35.925615 73059 util/stop/stopper.go:539  quiescing
E200127 19:26:35.934470 4605 jobs/registry.go:376  error while adopting jobs: failed querying for jobs: adopt-job: node unavailable; try another peer
W200127 19:26:35.936516 72679 internal/client/txn.go:566  [intExec=create-stats] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestImportCSVStmt PKG=./pkg/ccl/importccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestImportCSVStmt.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/jobs/registry.go
+++ b/pkg/jobs/registry.go
@@ -215,16 +215,17 @@ func (r *Registry) CreateAndStartJob(
 		return nil, nil, err
 	}
 	if err := j.insert(ctx, id, r.newLease()); err != nil {
-		r.unregister(*j.ID())
+		// Use id instead of job.ID because the new id is not recorded in the job.
+		r.unregister(id)
 		return nil, nil, err
 	}
 	if err := j.Started(ctx); err != nil {
-		r.unregister(*j.ID())
+		r.unregister(id)
 		return nil, nil, err
 	}
 	errCh, err := r.resume(resumeCtx, resumer, resultsCh, j)
 	if err != nil {
-		r.unregister(*j.ID())
+		r.unregister(id)
 		return nil, nil, err
 	}
 	return j, errCh, err

